### PR TITLE
Enable code coverage, check styles, and spot bugs

### PR DESCRIPTION
**Pull request by lahirusamith**

### CODE REVIEW

File: build-config/checkstyle/build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (BuddhiWathsala) [inline build-config/checkstyle/build.gradle:2]: We don't need to change the year if that file is already there. Please check.

File: build.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (BuddhiWathsala) [inline build.gradle:2]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: build-config/checkstyle/build.gradle (hunk 1 above)
Comment (BuddhiWathsala) [inline build-config/checkstyle/build.gradle:2]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: build-config/spotbugs-exclude.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+<FindBugsFilter>
+</FindBugsFilter>
```
Comment (BuddhiWathsala) [inline build-config/spotbugs-exclude.xml:19]:
```suggestion
</FindBugsFilter>

```

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: mysql-ballerina/build.gradle [hunk 4]
```diff
@@ -168,18 +178,31 @@ task ballerinaBuild {
...
     dependsOn(assemble)
     dependsOn(updateTomlFile)
+    dependsOn(initializeVariables)
     finalizedBy(revertTomlFile)
 
     inputs.dir file(project.projectDir)
+
+    def testParams = "--skip-tests"
+    gradle.taskGraph.whenReady { graph ->
+        if (graph.hasTask(':sql-ballerina:build')) {
```
Comment (BuddhiWathsala) [inline mysql-ballerina/build.gradle:188]:
```suggestion
        if (graph.hasTask(':mysql-ballerina:build')) {
```

File: mysql-ballerina/build.gradle [hunk 5]
```diff
@@ -168,18 +178,31 @@ task ballerinaBuild {
...
+            ballerinaTest.enabled = false
+        }
+        if (graph.hasTask(':sql-ballerina:test')) {
```
Comment (BuddhiWathsala) [inline mysql-ballerina/build.gradle:191]:
```suggestion
        if (graph.hasTask(':mysql-ballerina:test')) {
```

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request.yml [hunk 6]
```diff
@@ -42,10 +43,16 @@ jobs:
...
         with:
           name: Ballerina Internal Log Windows
           path: mysql-ballerina/ballerina-internal.log
           if-no-files-found: ignore
+      - name: Archive Code Coverage JSON
+        uses: actions/upload-artifact@v2
+        with:
+          name: Code Coverage JSON
+          path: mysql-ballerina/target/report/test_results.json
+          if-no-files-found: ignore
```
Comment (ThisaruGuruge) [inline .github/workflows/pull-request.yml:58]:
```suggestion
```

**Review by BuddhiWathsala — APPROVED** [review]

File: mysql-native/build.gradle [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (BuddhiWathsala) [inline mysql-native/build.gradle:2]:
```suggestion
 * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**